docs/dev.rst

```diff
@@ -32,29 +48,94 @@ These steps include-
-
-To speed up the search, AutoPyTorch and SMAC use `Dask.distributed <https://distributed.dask.org/en/latest/>`_ multiprocessing scheme. We only submits jobs to Dask.distributed.Client up to the number of workers, and wait for a worker to be available before continuing searching for more pipelines.
-
-At the end of SMAC, the results will be available in the `temporary_directory` provided to the API run, in particular inside of the `<temporary_directory>/smac3-output/run_<SEED>/` directory. One can debug
-the performance of the individual models using the file `runhistory.json` located in this area. Every individual model will be stored in `<temporary_directory>/.autoPyTorch/runs`. 
-In this `runs` directory we store the fitted model (during cross-validation we store a single Voting Classifier/Regressor, which is the soft voting outcome of k-Fold cross-validation), the Out-Of-Fold
-predictions that are used to build an ensemble, and also the test predictions of this model in question.
+In the case of tabular classification/regression,
+the training data is preprocessed using scikit-learn.compose.ColumnTransformer
+on a per-column basis.
+The data preprocessing is dynamically created depending on the dataset properties.
+For example, on a dataset that only contains float-type features,
+no one-hot encoding is needed.
+Additionally, we wrap the ColumnTransformer via TabularColumnTransformer class
+to support torchvision transformation and
+handle column-reordering.
+Note that column-reordering shifts categorical columns to the earlier indices
+and it is activated only if one uses a ColumnTransformer.
+
+Training of individual models
+-----------------------------
+
+Auto-PyTorch can fit 3 types of pipelines:
+
+#. Dummy pipeline: Use sklearn.dummy to construct an estimator that predicts using simple rules such as most frequent class
+#. Traditional machine learning pipelines: Use LightGBM, CatBoost, RandomForest, ExtraTrees, K-Nearest-Neighbors, and SupportVectorMachines
+#. PyTorch neural networks: Neural architecture search of backbones (feature extraction) and network heads (for the final prediction)
+
+Note that dummy pipeline is used as a baseline to define the worst-performing model that can be fit
+and traditional machine learning pipelines are critical for small-sized datasets.
+A pipeline might also contain additional training components
+like learning rate scheduler, optimizers,
+and data loaders required to perform the neural architecture search.
+
+After the training (fitting a pipeline), we use pickle to save it
+to disk as stated `here <https://scikit-learn.org/stable/modules/model_persistence.html>`_.
+
+Optimization of pipeline
+------------------------
+
+To optimize the pipeline, we use SMAC as mentioned earlier.
+Given a configuration, AutoPytorch fits a pipeline and 
+ finally saves to disc using the function evaluator `ExecuteTaFuncWithQueue`.
```

Comment:
Remove space before `finally` (it seems it makes this line a quotation)

Reply:
```suggestion
finally saves to disc using the function evaluator `ExecuteTaFuncWithQueue`.
```